lib/stdlib/src/gen_server.erl

```diff
@@ -1226,23 +1233,53 @@ format_status(Opt, StatusData) ->
 format_log_state(Mod, Log) ->
     [case Event of
          {out,Msg,From,State} ->
-             {out,Msg,From,format_status(terminate, Mod, get(), State)};
+             Status = limit_status(Mod, #{ opt => terminate,
+                                           state => State,
+                                           message => Msg }),
+             {out, maps:get(message, Status), From, maps:get(state, Status) };
          {noreply,State} ->
-             {noreply,format_status(terminate, Mod, get(), State)};
+             Status = limit_status(Mod, #{ opt => terminate,
+                                           state => State }),
+             {noreply, maps:get(state, Status)};
+         {in,Msg} ->
+             Status = limit_status(Mod, #{ opt => terminate,
+                                           message => Msg }),
+             {in, maps:get(message, Status)};
          _ -> Event
      end || Event <- Log].
 
-format_status(Opt, Mod, PDict, State) ->
-    DefStatus = case Opt of
-		    terminate -> State;
-		    _ -> [{data, [{"State", State}]}]
-		end,
-    case erlang:function_exported(Mod, format_status, 2) of
-	true ->
-	    case catch Mod:format_status(Opt, [PDict, State]) of
-		{'EXIT', _} -> DefStatus;
-		Else -> Else
-	    end;
-	_ ->
-	    DefStatus
+
+-type format_status() :: #{ opt := terminate | normal,
+                            state => term(),
+                            dictionary => [{term(),term()}],
+                            message => term() }.
+-spec limit_status(Mod :: module(), Status :: format_status()) -> format_status().
+limit_status(Mod, Status) ->
+    DefStatus =
+        case maps:get(opt, Status) of
+            terminate -> Status;
+            _ -> Status#{ state := [{data, [{"State", maps:get(state, Status, undefined)}]}]}
```

Comment:
Status  includes the copy of a process dictionary which could contain sensitive data.
In previous version of the code the `DefaultStatus` (on line 1238) didn't include PDict. Now it does.

Reply:
If you check [this line](https://github.com/erlang/otp/pull/4952/files#diff-1795d892ea0ec0b373b04026a3c449d436b22b1684eb5e69844efba22e86e305R1224), it is only the `state` that is actually returned when you do `sys:get_status`.

The `PDict` is included in the incoming status as it was available to the old implementation. Maybe I should just remove it from the `Status` as the user can always fetch it using `get()` themselves.